src/clj/game/cards/upgrades.clj

```diff
@@ -653,6 +653,31 @@
       :events {:pre-steal-cost ohg
                :post-access-card {:effect (effect (clear-persistent-flag! target :can-steal))}}})
 
+   "Overseer Matrix"
+   (let [om {:req (req (in-same-server? card target))
+             :delayed-completion true
+             :effect (effect (show-wait-prompt :runner "Corp to use Overseer Matrix")
+                             (continue-ability
+                               {:optional
+                                {:prompt "Pay 1 [Credits] to use Overseer Matrix ability?"
+                                 :player :corp
+                                 :end-effect (effect (clear-wait-prompt :runner)
+                                                     (effect-completed eid))
+                                 :yes-ability {:cost [:credit 1]
+                                               :msg "give the Runner 1 tag"
+                                               :delayed-completion true
+                                               :effect (req (tag-runner state :runner eid 1))}}}
+                               card nil))}]
+     {:trash-effect
+      {:req (req (and (= :servers (first (:previous-zone card))) (:run @state)))
```

Comment:
I don't believe a run is required for this effect, to account for trashing outside of runs.

Reply:
I think I do need the run requirement here. This is for when Overseer Matrix itself is trashed and we want the effect to continue for the remainder of the run. If it's trashed outside of a run, then we don't want to register the event handler.

When a `runner-trash` event happens, I don't check for a run.

Reply:
I forgot that `:runner-trash` triggers before the card moves zones, so we're good here.